### PR TITLE
ScoutSuite parser

### PR DESCRIPTION
**Pull request by hasantayyar**

### CODE REVIEW

File: dojo/templates/dojo/import_scan_results.html [hunk 1]
```diff
@@ -117,12 +117,14 @@ <h4> JIRA </h4>
 	<li><b>Risk Recon API Importer</b> - Risk Recon ApI will be accessed to gather finding information. Report format <a href="https://defectdojo.readthedocs.io/en/latest/integrations.html#risk-recon-api-importer">here</a>.</li>
 	<li><b>SARIF</b> - SARIF report file can be imported in SARIF format.</li>
 	<li><b>Safety Scan</b> - Safety scan (--json) output file can be imported in JSON format.</li>
-	<li><b>Semgrep</b> - Import Semgrep output (--json)</li>
+	<li><b>Safety Scan</b> - Safety scan (--json) output file can be imported in JSON format.</li>
```
Comment (damiencarol) [inline dojo/templates/dojo/import_scan_results.html:120]: Double copy/paste

Comment (hasantayyar) [reply dojo/templates/dojo/import_scan_results.html:120]: Thanks for quick lint.

**Comment by hasantayyar** [comment]
Sample scans and tests are going to be added and fixed.

**Comment by damiencarol** [comment]
@hasantayyar don't hesitate to use the feature of "Draft" => https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests

**Comment by valentijnscholten** [comment]
Thanks for the parser. Is there a reason why you are implementing the unit test differently from the existing unit tests for other parsers?

**Comment by hasantayyar** [comment]
> Thanks for the parser. Is there a reason why you are implementing the unit test differently from the existing unit tests for other parsers?

No reason. I just couldn't have time to prepare a scan result with "multiple vulnerabilities X multiple findings". So I imported only 2 versions of the scan.  One with no vulnerabilities and one with 2 vulnerabilities with 2 findings.

File: dojo/tools/scout_suite/parser.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+                               references=None,
+                               date=find_date,
+                               dynamic_finding=True)
+                dupes[dupe_key] = find
+        self.items = list(dupes.values())
+
+    def formatview(self, depth):
+        if depth > 1:
+            return "* "
+            print("depth hit")
```
Comment (damiencarol) [inline dojo/tools/scout_suite/parser.py:105]: Remove any print in the final version.

**Review by damiencarol — CHANGES_REQUESTED** [review]
some things need to be removed

File: dojo/tools/scout_suite/parser.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+# Cloned form aws_scout2 scanner
+from dojo.models import Finding
+from datetime import datetime
+import json
+from django.utils.text import Truncator
+from django.utils.html import strip_tags
+
+
+class ScoutSuiteParser(object):
+    item_data = ""
```
[the remark this answers is not in the document]
Comment (hasantayyar) [reply dojo/tools/scout_suite/parser.py:14]: Added wiki link.

**Comment by hasantayyar** [comment]
> some things need to be removed

like what?

**Comment by hasantayyar** [comment]
> some things need to be removed

@damiencarol 

like what?

File: dojo/tools/scout_suite/parser.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        test_description = "%s\n**Services** \n\n" % (test_description)
+        for service, items in list(last_run["summary"].items()):
+            test_description = "%s\n**%s** \n" % (test_description, service.upper())
+            test_description = "%s\n* **Checked Items:** %s\n" % (test_description, items["checked_items"])
+            test_description = "%s* **Flagged Items:** %s\n" % (test_description, items["flagged_items"])
+            test_description = "%s* **Max Level:** %s\n" % (test_description, items["max_level"])
+            test_description = "%s* **Resource Count:** %s\n" % (test_description, items["resources_count"])
+            test_description = "%s* **Rules Count:** %s\n\n" % (test_description, items["rules_count"])
+        test.description = test_description
+        test.save()
```
Comment (damiencarol) [inline dojo/tools/scout_suite/parser.py:42]: Could you get ride of `.save()`. it's too low level here.

File: dojo/tools/scout_suite/parser.py [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+        # Summary of Services
+        test_description = "%s\n**Services** \n\n" % (test_description)
+        for service, items in list(last_run["summary"].items()):
+            test_description = "%s\n**%s** \n" % (test_description, service.upper())
+            test_description = "%s\n* **Checked Items:** %s\n" % (test_description, items["checked_items"])
+            test_description = "%s* **Flagged Items:** %s\n" % (test_description, items["flagged_items"])
+            test_description = "%s* **Max Level:** %s\n" % (test_description, items["max_level"])
+            test_description = "%s* **Resource Count:** %s\n" % (test_description, items["resources_count"])
+            test_description = "%s* **Rules Count:** %s\n\n" % (test_description, items["rules_count"])
+        test.description = test_description
```
Comment (damiencarol) [inline dojo/tools/scout_suite/parser.py:41]: This part is really good. You are providing more data aligned with the source. It's good!

File: dojo/tools/scout_suite/parser.py [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+            title = strip_tags(scout2_finding["title"])
+            sev = self.getCriticalityRating(scout2_finding["severity"])
+            description = scout2_finding["description"]
+            dupe_key = sev + title
+            if dupe_key in dupes:
+                find = dupes[dupe_key]
+                if description is not None:
+                    find.description += description
+            else:
+                find = Finding(title=Truncator(title).words(6),
```
Comment (damiencarol) [inline dojo/tools/scout_suite/parser.py:89]: If you cut it, could be good to add "..." to the string

File: dojo/tools/scout_suite/parser.py [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+                               cwe=1032,  # Security Configuration Weaknesses, would like to fine tune
+                               test=test,
+                               active=False,
+                               verified=False,
+                               description="**Account:** " + account_id + "\n" + description,
+                               severity=sev,
+                               numerical_severity=Finding.get_numerical_severity(sev),
+                               references=None,
```
Comment (damiencarol) [inline dojo/tools/scout_suite/parser.py:97]: It seems that there is references data in the data, don't you want to use it?

```JSON
           "iam-role-assigned-to-user":{
               "checked_items":0,
               "compliance":null,
               <...>
               "references":[
                  "https://cloud.google.com/iam/docs/understanding-roles",
                  "https://cloud.google.com/iam/docs/using-iam-securely"
               ],
               <...>
               "remediation":null,
               "service":"IAM"
            },
```

File: dojo/tools/scout_suite/parser.py (hunk 4 above)
Comment (damiencarol) [reply dojo/tools/scout_suite/parser.py:42]:
```suggestion
```

**Review by damiencarol — CHANGES_REQUESTED** [review]

File: dojo/tools/scout_suite/parser.py (hunk 5 above)
Comment (hasantayyar) [reply dojo/tools/scout_suite/parser.py:41]: Actually it is cloned from aws_scout parser as noted at the top.

File: dojo/tools/scout_suite/parser.py (hunk 6 above)
Comment (hasantayyar) [reply dojo/tools/scout_suite/parser.py:89]: Truncator does it automatically.
https://docs.djangoproject.com/en/1.8/_modules/django/utils/text/

File: dojo/tools/scout_suite/parser.py (hunk 7 above)
Comment (hasantayyar) [reply dojo/tools/scout_suite/parser.py]: Good idea. added

**Comment by hasantayyar** [comment]
@damiencarol  
I did not understand the reason at this change request https://github.com/DefectDojo/django-DefectDojo/pull/3602#discussion_r555012612

File: dojo/tools/scout_suite/parser.py (hunk 6 above)
Comment (damiencarol) [reply dojo/tools/scout_suite/parser.py:90]: oh

File: dojo/tools/scout_suite/parser.py [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+                                "description": description_text,
+                                "references": finding["references"]
+                            }
+                            scoutsuite_findings.append(mobsf_item)
+
+        for scoutsuite_finding in scoutsuite_findings:
+            title = strip_tags(scoutsuite_finding["title"])
+            sev = self.getCriticalityRating(scoutsuite_finding["severity"])
+            description = scoutsuite_finding["description"]
+            references = scoutsuite_finding["references"]
```
Comment (damiencarol) [inline dojo/tools/scout_suite/parser.py:83]: @hasantayyar the `references` attributes is a simple Textfield. We need to convert it to a string I think.

Comment (hasantayyar) [reply dojo/tools/scout_suite/parser.py:83]: Right.

It is being auto converted now

image

Better to convert to string.

Comment (hasantayyar) [reply dojo/tools/scout_suite/parser.py:84]: Pushed the change. 
References is an array in the scan report. So I safely joined the array with whitespace

**Review by damiencarol — APPROVED** [review]
Thanks for your hard work @hasantayyar ! It's a good PR!

**Comment by valentijnscholten** [comment]
Just noticed the PR is based against `master`. Could you rebase against / onto `dev`? the base branch then can be changed here on the PR also to dev.

**Comment by hasantayyar** [comment]
> Just noticed the PR is based against `master`. Could you rebase against / onto `dev`? the base branch then can be changed here on the PR also to dev.


Rebased.

File: .github/renovate.json [hunk 9]
```diff
@@ -3,7 +3,7 @@
     "config:base"
   ],
   "dependencyDashboard": true,
-  "dependencyDashboardApproval": true,
+  "dependencyDashboardApproval": false,
```
Comment (damiencarol) [inline .github/renovate.json:6]:
```suggestion
  "dependencyDashboardApproval": true,
```

**Review by damiencarol — CHANGES_REQUESTED** [review]
You need to add a modification to the file `dojo\fixtures\test_type.json`

File: .github/renovate.json (hunk 9 above)
Comment (hasantayyar) [reply .github/renovate.json]: Also that was added by  @valentijnscholten here https://github.com/DefectDojo/django-DefectDojo/pull/3602/commits/d09d3e3f48ec5f6ad3906a2f63a3b42181f3569b

File: .github/renovate.json [hunk 10]
```diff
@@ -3,7 +3,7 @@
     "config:base"
   ],
   "dependencyDashboard": true,
-  "dependencyDashboardApproval": false,
+  "dependencyDashboardApproval": true,
```
Comment (valentijnscholten) [inline .github/renovate.json:6]: this needs to stay false.

File: dojo/fixtures/test_type.json [hunk 11]
```diff
@@ -698,6 +698,13 @@
         },
         "model": "dojo.test_type",
         "pk": 191
+    },
```
Comment (valentijnscholten) [inline dojo/fixtures/test_type.json:701]: this needs an update since we merged 2 other parsers today, sorry.

**Review by valentijnscholten — CHANGES_REQUESTED** [review]
see comments

File: .github/renovate.json (hunk 10 above)
Comment (hasantayyar) [reply .github/renovate.json]: @damiencarol FYI

**Comment by hasantayyar** [comment]
Rebased

**Comment by hasantayyar** [comment]
@valentijnscholten test number is incremented 

@damiencarol I reverted `dependencyDashboardApproval` to false again according to comment from @valentijnscholten

**Review by damiencarol — APPROVED** [review]

**Comment by damiencarol** [comment]
@hasantayyar Could you make also a PR to add a scan to https://github.com/DefectDojo/sample-scan-files ?

**Comment by hasantayyar** [comment]
> @hasantayyar Could you make also a PR to add documentation to https://github.com/DefectDojo/Documentation/blob/master/docs/integrations.rst ?

Added https://github.com/DefectDojo/Documentation/pull/169 
@damiencarol

**Comment by hasantayyar** [comment]
merged dev and fixed conflicts

**Comment by hasantayyar** [comment]
@valentijnscholten Can you please re-review

**Review by damiencarol — APPROVED** [review]

**Review by valentijnscholten — APPROVED** [review]

**Comment by valentijnscholten** [comment]
@hasantayyar can you add a sample scan to our repo with sample scans?

**Comment by hasantayyar** [comment]
> @hasantayyar can you add a sample scan to our repo with sample scans?

@valentijnscholten You mean this https://github.com/DefectDojo/sample-scan-files/pull/64 

or smething else?

**Comment by valentijnscholten** [comment]
Yes thanks

**Pull request merged**